src/server/game/Entities/Object/Object.h

```diff
@@ -436,7 +436,7 @@ class TC_GAME_API WorldObject : public Object, public WorldLocation
         virtual void RemoveFromWorld() override;
 
         void GetNearPoint2D(float &x, float &y, float distance, float absAngle) const;
-        void GetNearPoint(WorldObject const* searcher, float &x, float &y, float &z, float searcher_size, float distance2d, float absAngle) const;
+        void GetNearPoint(WorldObject const* searcher, float &x, float &y, float &z, float searcher_size, float distance2d, float absAngle, bool is3D = false) const;
```

Comment:
I'm not sure this is the right API.

Given we have GetNearPoint2D, the clearest thing to me, would be to have a GetNearPoint3D, and avoid the whole bool-flag thing.

In general, [rules](http://www.planetgeek.ch/wp-content/uploads/2014/11/Clean-Code-V2.4.pdf) of thumb should be 
 * No flag arguments in functions 
 * Method with too many arguments